plugins/processors/t128_filter/t128_filter.go

```diff
@@ -0,0 +1,129 @@
+package t128_filter
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const sampleConfig = `
+[[processors.t128_filter]]
+  ## The conditions that must be met to pass a metric through. This is similar
+  ## behavior to a tagpass, but the multiple tags are ANDed
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+     #tag1 = ["value1", "value2"]
+	 #tag2 = ["value3"]
+`
+
+type tags map[string][]string
+
+type Condition struct {
+	Tags tags `toml:"tags"`
+}
+
+type T128Filter struct {
+	Conditions []Condition `toml:"condition"`
+
+	log     telegraf.Logger `toml:"-"`
+	matcher matcher         `toml:"-"`
+}
+
+func (r *T128Filter) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *T128Filter) Description() string {
+	return "Filter metrics from being emitted."
+}
+
+type matcher interface {
+	Matches(telegraf.Metric) bool
+}
+
+type exactMatcher struct {
+	tag    string
+	values []string
+}
+
+func (m exactMatcher) Matches(point telegraf.Metric) bool {
+	value, ok := point.GetTag(m.tag)
+	if !ok {
+		return false
+	}
+
+	for _, expectedValue := range m.values {
+		if value == expectedValue {
+			return true
+		}
+	}
+
+	return false
+}
+
+type andConjMatcher struct {
+	matchers []matcher
+}
+
+func (c andConjMatcher) Matches(point telegraf.Metric) bool {
+	for _, matcher := range c.matchers {
+		if !matcher.Matches(point) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func createMatcher(conditions []Condition) (matcher, error) {
+	conditionMatchers := make([]matcher, len(conditions))
+	for i, condition := range conditions {
+		tagMatchers := getTagMatchers(condition.Tags)
+
+		conditionMatchers[i] = andConjMatcher{matchers: tagMatchers}
+	}
+
+	return andConjMatcher{conditionMatchers}, nil
+}
+
+func getTagMatchers(tags tags) []matcher {
+	tagMatchers := make([]matcher, len(tags))
+
+	j := 0
+	for tagKey, tagValues := range tags {
+		tagMatchers[j] = exactMatcher{tag: tagKey, values: tagValues}
+		j++
+	}
+
+	return tagMatchers
+}
+
+func (r *T128Filter) Apply(in ...telegraf.Metric) []telegraf.Metric {
```

Comment:
the tagpass/tagdrop gets evaluted by telegraf code, correct? So we only get the specific rows that we care about in that case?

Reply:
Correct